hydromt/models/model.py

```diff
@@ -94,21 +108,15 @@ def __init__(
         logger:
             The logger to be used.
         """
+        # Recursively update the options with any defaults that are missing in the configuration.
+        components = components or {}
+        components = pydantic.v1.utils.deep_update(
+            {"region": {"type": "ModelRegionComponent"}}, components
+        )
+
         data_libs = data_libs or []
-        from . import MODELS  # avoid circular import
 
         self.logger = logger
-        dist_name, version = "unknown", "NA"
-        if self._NAME in MODELS:
-            ep = MODELS[self._NAME]
-            dist: Optional[Distribution] = ep.dist
-            if dist:
-                dist_name = dist.name
-                version = dist.version
-            else:
-                # insert hydromt defaults
-                dist_name = "hydromt"
-                version = __version__
```

Comment:
I think we need to put something like this in place again, but I don't know if the initialization of Model is the right place.

Reply:
I removed this during the plugin refactor because it was causing problems, and planned to add it back during the config component implementation, but I can do this now if you want